### DISCUSSION TITLE
Отличная работа на Mikrotik RB750Gr3

### DISCUSSION THREAD

**Discussion by BubaMuba** [bol-van/zapret#300] [General]
Openwrt 23.05.4 на Mikrotik RB750Gr3 в режиме nfqws c флагами --dpi-desync=fake,split2 --dpi-desync-ttl=2
Использую пустой хостлист, в exclude буквально пара исключений. Семейство довольно, работает всё, и работает быстро.
Полагаю, что будет справедливо задонатить автору за это волшебство. Реквизитов только не нашёл. Много не обещаю, но чем смогу.

**Comment by bol-van** [comment]
если только биток
номера карт и телефон постить не буду даже в личку

**Reply by BubaMuba** [reply]
С битком как-то всё очень заморочено, и с законом могут быть неприятности. Если решитесь на банковский перевод, дайте знать.

**Reply by bol-van** [reply]
полно обменников или биржа yobit.net
если там не миллион, проблем нет.

**Reply by BubaMuba** [reply]
Ну я попытался закинуть деньги через yobit.net
Там вообще какое-то наебалово. Сорри, но это самое подходящее слово. Деньги надо перевести на карту Равшана, а по факту он Джамшут.
Вот если честно, мне сегодня хотелось сделать доброе дело, но уже перехотелось.

**Reply by bol-van** [reply]
тогда ничего не надо делать

**Comment by 777Evgeny777** [comment]
Добрый день.
Автору респект.
BubaMuba,подскажите пожалуйста , по какой инструкции на микротик ставили, у меня не получается поставить на RBmap2nd, с флэшкой, ругается на iptables, какие пакеты необходимо установить ?

**Reply by BubaMuba** [reply]
Устанавливал по инструкции из quick_start.txt.
Если у вас, как и у меня, OpenWrt 23.05.4, то iptables не нужен. Там fw4 и nftables. Соответственно, при запуске install_easy.sh выбирайте nftables.

**Comment by 777Evgeny777** [comment]
Спасибо, необходимые пакеты сами установятся или надо их сначала установить перед выполнением инструкции ?
со знаниями оперврт бог обидел, потому прошу заранее извинить за такие вопросы

**Reply by BubaMuba** [reply]
Для работы zapret'a установятся сами.

**Comment by 777Evgeny777** [comment]
Спасибо, заработало, до перезагрузки роутера, после перезагрузки перестало, буду разбираться.

**Comment by 777Evgeny777** [comment]
Спасибо, заработало, до перезагрузки роутера, после перезагрузки перестало, буду разбираться.

**Reply by BubaMuba** [reply]
Почитайте вывод команды logread. Там бывает полезная информация.

**Comment by 777Evgeny777** [comment]
Уже разобрался, флэшка Кингстон не определялась, только после передергивания, поставил другую начала нормально монтироваться сама

**Comment by 777Evgeny777** [comment]
Единственно не смог настроить автолист, не попадают в него домены и если вручную заносишь не открываются, поразбираюсь, заметил что некоторые пути не совпадают с инструкциями, может дело в этом. 
интересно на опенврт 19 можно это заставить работать ?

**Reply by BubaMuba** [reply]
Я с автолистом побаловался немного, меня он тоже не устроил. Как написал в первом сообщении, настроил работу с hostlist'ом, при этом сделал его пустым и весь трафик https заворачивается на nfqws. В этом режиме работает 10 дней, никаких косяков не выявлено, а ютуб просто летает.
Полагаю, что на 19 тоже всё будет нормально, только там выбирайте iptables.

**Reply by bol-van** [reply]
у автолиста есть дебаг лог

**Comment by 777Evgeny777** [comment]
iptables+nfqws ?

вон на 19 версии у меня тоже  были грабли с iptables, выдавал ошибки, сейчас уже не вспомню какие, может попробую воспроизвести

**Reply by BubaMuba** [reply]
Если вы блокчеком нашли подходящуюю стратегию, то на 19 openwrt используйте её же + iptables.
Ещё лучше, если предварительно прочитаете docs/manual_setup.txt, раздел Ручная установка на openwrt/LEDE 15.xx-21.xx. И совсем хорошо, если прочитаете всю предоставленную автором документацию.

**Reply by bol-van** [reply]
если есть какие-то проблемы с iptables на этих openwrt, пишите конкретику
у меня один из роутеров openwrt 19.07. работает

**Comment by 777Evgeny777** [comment]
интересно в метароутере этот шедевр запустить удастся ?

**Reply by BubaMuba** [reply]
Я не пробовал, благо на мою железку openwrt влетает без проблем.
Если получится, поделитесь информацией.

**Comment by 777Evgeny777** [comment]
я нашел только опенврт 12 на метароутер, моих знаний наверно не хватит, чтобы вышло что-то толковое

**Reply by BubaMuba** [reply]
Знания - дело наживное. Но лично я не стал бы заморачиваться с такими костылями. Если переживаете за RouterOS, то напрасно, OpenWRT не хуже. А для многих задач, в том числе этой - лучше.

**Comment by 777Evgeny777** [comment]
я не нашел как управлять POE выходом портов на опенвпн

**Reply by BubaMuba** [reply]
Не openvpn, openwrt. Так пробовали?:
`uci set system.poe_passthrough.value=1`
`uci commit`

**Comment by 777Evgeny777** [comment]
Спасибо, попробую,
В данный момент пробую поставить опенврт 19 на RB750UPR2, не монтируется флэшка, в логе пишет
 user.err kernel: [  110.498906] block: extroot: cannot find device with UUID ef4ea379-2d12-424c-a91a-e4c1fbaf16db
хотя идентификатор верный, по индикатору питания флэшки сделал вывод , что питание на порт USB подается позднее чем система пытается найти флэш, пока на этом застопорился, еще помучаюсь и попробую на RB951 установить 19 версию со всеми рекомендациями данными выше

**Reply by BubaMuba** [reply]
Посмотрите это: https://openwrt.org/docs/guide-user/additional-software/extroot_configuration
И это: https://github.com/openwrt/openwrt/issues/10045
Но по большому счету, extroot на этих микротиках для zapret'a не обязателен, места и так хватает.

**Comment by 777Evgeny777** [comment]
Спасибо, посмотрю, справился с установкой шедевра, без флэшки, памяти хватило впритык на RB750UPR2, но все работает

**Reply by BubaMuba** [reply]
Почему впритык-то? Там же 16 Mb, по факту, наверное, около 10-11 с чистой openwrt. Этого хватает и на zapret, и на tcpdump, и на всякие nano, bmon'ы и htop'ы.

**Reply by bol-van** [reply]
Может быть копировал всю директорию запрета в /opt ?
Без инсталятора и отсечения лишних файлов ?

**Reply by 777Evgeny777** [reply]
Я только учусь, возможно лишнего осталось, переустанвливаю сейчас, Люси сдулась, удалил что-то лишнее, но сам факт показал что на опенврт 19 работает, буду пробовать накатить на самую чахлую железку RB941

**Reply by 777Evgeny777** [reply]
Пока нет четкого понимания какие пакеты нужны, в разных местах по разному написано, какие-то вообще не ставятся под 19 оврт, иду путем проб и ошибок, пока позволяет наличие свободного времени.

**Reply by bol-van** [reply]
Инсталятор сам доустановит нужные пакеты, если их нет

**Reply by 777Evgeny777** [reply]
не хватает памяти если использовать инсталлятор, на http-get пишет что места нет

**Reply by 777Evgeny777** [reply]
я ставлю оврт19, потом доустанавливаю пакеты для запуска самого инсталлятора, иначе он не запускается

**Reply by 777Evgeny777** [reply]
сейчас выведу список пакетов с чистой оврт19 (установлю на RB941 если быстро получится), подскажите пожалуйста  что надо поставить чтобы инсталлятор запустился

**Reply by bol-van** [reply]
Если места нет - это не поможет
Оригинал комплекта zapret надо в /tmp. Он в RAM

**Reply by 777Evgeny777** [reply]
Ок, так и делаю

**Reply by bol-van** [reply]
Если памяти совсем мало, см manual_setup.txt
Может потребоваться отказаться от скриптов запуска и руками все делать

Еще можно снести files/fake, если они не используются
blockcheck.sh, install_prepreq.sh тоже не нужен все время
Можно освободить 150 кб

**Reply by BubaMuba** [reply]
Совсем простая инструкция:
1. На PC с линуксом: 
`~$ git clone --depth 1 https://github.com/bol-van/zapret`
2. На PC с линуксом: 
`~$ scp -r zapret root@192.168.1.1:/tmp/`
3. На PC с линуксом: 
`~$ ssh root@192.168.1.1`
4. На железке с openwrt:
`~# cd /tmp/zapret`
Далее запускайте install_bin.sh, install_prereq.sh и т.д. по инструкции.
Если на железке другой ip, то подставьте его вместо 192.168.1.1

**Reply by BubaMuba** [reply]
Когда установите zapret инсталлятором в /opt, проверьте свободное место командой `df -h`, и удивитесь, сколько всего ещё можно установить.

**Reply by bol-van** [reply]
Я бы советовал перед scp удалить .git. Он может раздуть размер до 20 мб вместо 6.
Или вместо git clone качать ZIP с гитхаба

**Comment by 777Evgeny777** [comment]
Спасибо, за ценную информацию, я оценил, думаю людям тоже понадобится.
на RB941 owrt19.07.10 ставиться не хочет, после заливки образа в RAM и его запуска, зависает при попытке залить sysupgrade во флэшь, пока эксперименты с чахлыми железками откладываю по причине отсутствия времени(много времени уходит на многократную подготовку железки , залития образа  и т.п.), результат на более-менее нормальных железках Mikrotik положительный.RB951 летает нормально (памяти вагон), RB2011 тоже все норм, RBMap2nd работает (с внешней флэшкой), но у верен можно и без нее, RB750UPR2 не смог заставить работать на внешней флэшке, но и без нее запустилось. Автору и соратникам огромный респект за труды ,терпение и желание нести знания в массы !!!!
ROS - вещь !!!
OPENWRT - сила !!!

**Comment by 777Evgeny777** [comment]
Вопрос у меня остался при запуске на owrt19.07.10 файла install_prereq выходит ошибка установки пакета :
Unknown package 'kmod-nft-queue'.
Collected errors:
opkg_install_cmd: Cannot install package kmod-nft-queue.
could not install prerequisites

можно каким то пакетом заменить или придется c iptable работать ?

**Reply by BubaMuba** [reply]
На первом вопросе выбирайте iptables

**Reply by 777Evgeny777** [reply]
ну это я уже понял

**Reply by bol-van** [reply]
nftables поддерживаются начиная с openwrt 22
в более старых openwrt даже если они и есть, то старые и несовместимые

**Comment by 777Evgeny777** [comment]
вот так обстоят дела с памятью на RB750 сразу после установки всех указанный в руководстве пакетов и самого Zapret :
под оверлее места не остается, даже МС туда не поставить
root@OpenWrt:~# df
Filesystem           1K-blocks      Used Available Use% Mounted on
/dev/root                 2560      2560         0 100% /rom
tmpfs                    29696        84     29612   0% /tmp
/dev/mtdblock8           12032     11852       180  99% /overlay
overlayfs:/overlay       12032     11852       180  99% /
tmpfs                      512         0       512   0% /dev

Но все работает.

**Reply by BubaMuba** [reply]
Не густо с местом, да. Тогда лучше сделать extroot, или хотя бы /opt вынести на флешку.

**Reply by 777Evgeny777** [reply]
это просто эксперименты по запуску , нет задачи во что бы то ни стало юзать железку еще какой то задачей, для себя выясняю предел по железу, что бы потом понимать какое нужно для минимального запуска и дальнейшей работы не только в качестве мясорубки

**Reply by bol-van** [reply]
180 кб из 12 мб - ищите чем забито место. места вагон
ncdu может помочь

**Comment by BubaMuba** [comment]
Отчитываюсь после двух месяцев работы.
Запрет был поднят на двух одинаковых устройствах (Mikrotik RB750Gr3). На первом Openwrt 23.05.4 c extroot на microsd. На втором Openwrt 23.05.5 на родной nand 16 Mb. После трех недель активного юзания (обновлений запрета и изменений конфигов) вторая железка заглючила в работе, после перезагрузки не поднялась. Удалось снова накатить на неё openwrt, сейчас буду тоже делать extroot. Первое устройство работает изумительно прекрасно.
P.S. Пользуясь случаем, решил попробовать вернуть RouterOS на кончившийся микрот. Не получилось, netinstall просто ни в какую его не видит. Видимо, это билет в один конец, поэтому подумайте, прежде чем накатывать openwrt.

**Comment by 777Evgeny777** [comment]
Тоже напишу
Работает на 23.05.4, 23.05.5, 19.07.10
на RB951G-2Hnd, RB951UI-2Hnd, RB941, RB750UP, RB750UPR2
на родную прошивку возвращаются нормально, тоже были проблемы с нетинсталлом , но победил, даже лицухи не слетели, нужно подключать роутер к компу через свитч, чтобы порт не падал при перезагрузке , ну и соблюсти требования, помогает включение/выключение брандмауэра, антивиря, можно попробовать на другом компе,если что ОСь win7x64
